redfish-core/lib/rde.hpp

```diff
@@ -48,32 +50,191 @@ struct Resource
+        auto uriIt = uriMap.find(id);
+        if (uriIt != uriMap.end())
+        {
+            oss2 << uriIt->second;
+        }
+        else
+        {
+            oss2 << it->str(); // leave unchanged
+        }
+        lastPos2 = static_cast<size_t>(it->position() + it->length());
+    }
+    oss2 << bejJson.substr(lastPos2);
+    bejJson = oss2.str();
+
+    bejJson = std::regex_replace(bejJson, std::regex(R"(%I(\d+))"), "$1");
+
+    return nlohmann::json::parse(bejJson);
 }
 
-// PATCH handler
-inline void rdePatchHandler(App& app, const crow::Request& req,
-                            const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+inline bool rdeHandleCreateTask(const boost::system::error_code& ec2,
+                                sdbusplus::message_t& msg,
+                                const std::shared_ptr<task::TaskData>& taskData)
 {
-    BMCWEB_LOG_DEBUG("RDE : Patch handler.");
+    if (ec2)
+    {
+        return task::completed;
+    }
 
-    if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+    dbus::utility::DBusPropertiesMap changedProperties;
+    msg.read(changedProperties);
+
+    const std::string* data = nullptr;
+    const uint16_t* rc = nullptr;
+    nlohmann::json taskEntry;
+
+    for (const auto& propertyPair : changedProperties)
     {
-        return;
+        if (propertyPair.first == "payload")
+        {
+            data = std::get_if<std::string>(&propertyPair.second);
+            const auto& updatedData = handleDeferredBindings(*data);
+            taskEntry["Payload"] = updatedData;
+        }
+        else if (propertyPair.first == "return_code")
+        {
+            rc = std::get_if<uint16_t>(&propertyPair.second);
+            taskEntry["ReturnCode"] = *rc;
+        }
+    }
+    if (data == nullptr || rc == nullptr)
+    {
+        BMCWEB_LOG_ERROR("Invalid property or property not found");
+        taskData->messages.emplace_back(messages::internalError());
+        return task::completed;
     }
 
+    std::string index = std::to_string(taskData->index);
+    taskData->messages.emplace_back(taskEntry);
+    taskData->messages.emplace_back(messages::taskCompletedOK(index));
+    taskData->state = "Completed";
+
+    return task::completed;
+}
+
+inline void rdeCreateTask(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                          task::Payload&& payload,
+                          const sdbusplus::message::object_path& objPath)
+{
+    std::string matchString =
+        ("type='signal',"
+         "path='" +
+         objPath.str +
+         "',"
+         "interface='xyz.openbmc_project.RDE.OperationTask',"
+         "member='TaskUpdated'");
+
+    std::shared_ptr<task::TaskData> task = task::TaskData::createTask(
+        std::bind_front(rdeHandleCreateTask), matchString);
+    task->startTimer(std::chrono::minutes(5));
+    task->populateResp(asyncResp->res);
+    task->payload.emplace(std::move(payload));
+}
+
+inline uint32_t getOperationID()
```

Comment:
This is unique from process point. incase bmcweb restarts with some erason , agin reset to 1 . and if we use same operation ID RDE device fails. Need toi add TODO to handle in better way , like persistent approch or something else.